test/main/utilities.js

```diff
@@ -0,0 +1,43 @@
+QUnit.module( "QUnit.objectType" );
+
+function buildFunctions() {
+	var objects = [ function( ) { } ];
+
+	try {
+		var arrow = new Function( "return () => {};" )( );
+		objects.push( arrow );
+	} catch ( e ) { }
+
+	try {
+		var async = new Function( "return async function( ) { };" )( );
+		objects.push( async );
+	} catch ( e ) { }
```

Comment:
@leobalter - The same tests are ran on IE, Phantom, possibly Node 6, etc where `async function() { }` would be a parse error. The intent of the `try` / `catch` is to ensure that when `async function() { }` is not valid syntax, the tests for the plain function still runs.